[SEC-32636] docs(mcp): add IoC Explorer tools to MCP Server tools page (#37500)
DataDog/documentation: content/en/mcp_server/tools.md

## Security

> Tools for code security scanning, analyzing, searching and triaging [security signals][53], investigating [IoC Explorer][64] indicators, managing [detection rules][60] and [suppressions][61], and analyzing [security findings][54].

Reviewer: ```suggestion
Tools for code security scanning, analyzing, searching, and triaging [security signals][53], investigating [IoC Explorer][64] indicators, managing [detection rules][60] and [suppressions][61], and analyzing [security findings][54].
```

- Set all open signals for `service:checkout` to under review and assign them to me.
- Mark signal `AwAAAZ27F1BUjY4rPQAAABhBWjI3RjFCVWpZNHJBQUFBSGFNQVZBQUFBR1Bu` as archived with reason "testing".

### `get_datadog_ioc_filter_values`
*Toolset: **security***\
*Permissions Required: `Security Signals Read`*\
Discover filterable fields and their values for [IoC Explorer][64]. Omit `filter` to list available fields; supply `filter` to get `[{value, count}]` for that field. Use `query` to scope counts to a subset of indicators.

> - What fields can I filter IoC indicators by?

Reviewer: ```suggestion
- What fields are available for IoC indicator filters?
```

Author: applied both!
